radio/src/targets/common/arm/stm32/aux_serial_driver.cpp

```diff
@@ -257,142 +231,169 @@ extern "C" void AUX_SERIAL_USART_IRQHandler(void)
     DMA_InitStructure.DMA_DIR = DMA_DIR_PeripheralSRC;
-    DMA_InitStructure.DMA_BufferSize = aux2SerialRxFifo.size();
+    DMA_InitStructure.DMA_BufferSize = aux4SerialRxFifo.size();
     DMA_InitStructure.DMA_PeripheralInc = DMA_PeripheralInc_Disable;
     DMA_InitStructure.DMA_MemoryInc = DMA_MemoryInc_Enable;
     DMA_InitStructure.DMA_PeripheralDataSize = DMA_PeripheralDataSize_Byte;
     DMA_InitStructure.DMA_MemoryDataSize = DMA_MemoryDataSize_Byte;
     DMA_InitStructure.DMA_Mode = DMA_Mode_Circular;
     DMA_InitStructure.DMA_Priority = DMA_Priority_Low;
     DMA_InitStructure.DMA_M2M = DMA_M2M_Disable;
-    DMA_Init(AUX2_SERIAL_DMA_Channel_RX, &DMA_InitStructure);
-    USART_DMACmd(AUX2_SERIAL_USART, USART_DMAReq_Rx, ENABLE);
-    USART_Cmd(AUX2_SERIAL_USART, ENABLE);
-    DMA_Cmd(AUX2_SERIAL_DMA_Channel_RX, ENABLE);
-
-    // AUX2_SERIAL_USART->CR1 |= USART_CR1_RXNEIE; // USART_ITConfig(AUX2_SERIAL_USART, USART_IT_RXNE, ENABLE);
-
-  // else {
-  //   USART_Cmd(AUX2_SERIAL_USART, ENABLE);
-  //   USART_ITConfig(AUX2_SERIAL_USART, USART_IT_RXNE, ENABLE);
-    // USART_ITConfig(AUX2_SERIAL_USART, USART_IT_TXE, DISABLE);
-    NVIC_SetPriority(AUX2_SERIAL_USART_IRQn, 7);
-    NVIC_EnableIRQ(AUX2_SERIAL_USART_IRQn);
-  // }
-}
-
-void aux2SerialInit(void)
-{
-//   aux2SerialStop();
-
-//   aux2SerialMode = mode;
+    DMA_Init(AUX4_SERIAL_DMA_Channel_RX, &DMA_InitStructure);
+    USART_DMACmd(AUX4_SERIAL_USART, USART_DMAReq_Rx, ENABLE);
+    USART_Cmd(AUX4_SERIAL_USART, ENABLE);
+    DMA_Cmd(AUX4_SERIAL_DMA_Channel_RX, ENABLE);
 
-  aux2SerialSetup(AUX2_SERIAL_BAUDRATE, true);
+    NVIC_SetPriority(AUX34_SERIAL_USART_IRQn, 7);
+    NVIC_EnableIRQ(AUX34_SERIAL_USART_IRQn);
 }
 
-void aux2SerialPutc(char c)
+void aux4SerialInit(void)
 {
-#if !defined(SIMU)
-  if (aux2SerialTxFifo.isFull()) return;
-
-  aux2SerialTxFifo.push(c);
-  USART_ITConfig(AUX2_SERIAL_USART, USART_IT_TXE, ENABLE);
-#endif
+  aux4SerialSetup(AUX4_SERIAL_BAUDRATE, true);
 }
 
-void aux2SerialStop(void)
+void aux4SerialStop(void)
 {
-#if defined(STM32F0)
-  DMA_DeInit(AUX2_SERIAL_DMA_Channel_RX);
-#else
-  DMA_DeInit(AUX2_SERIAL_DMA_Stream_RX);
-#endif
-  USART_DeInit(AUX2_SERIAL_USART);
+  DMA_DeInit(AUX4_SERIAL_DMA_Channel_RX);
+  USART_DeInit(AUX4_SERIAL_USART);
 }
 
-void aux2SerialSetIdleCb(void (*cb)()) {
-  if (cb == nullptr) {
-    AUX2_SERIAL_USART->CR1 &= ~USART_CR1_IDLEIE;
-  } else {
-    AUX2_SERIAL_USART->CR1 |= USART_CR1_IDLEIE;
-  }
-  aux2SerialIdleCb = cb;
+void aux4SerialSetIdleCb(void (*cb)()) {
+  aux4SerialIdleCb = cb;
+  AUX4_SERIAL_USART->CR1 |= USART_CR1_IDLEIE;
 }
+#endif // AUX4_SERIAL
 
+#if defined(AUX3_SERIAL) || defined(AUX4_SERIAL)
 #if !defined(SIMU)
-extern "C" void AUX2_SERIAL_USART_IRQHandler(void)
+extern "C" void AUX34_SERIAL_USART_IRQHandler(void)
 {
   // Send
-  if (USART_GetITStatus(AUX2_SERIAL_USART, USART_IT_TXE) != RESET) {
+#if defined(AUX3_SERIAL_TX)
```

Comment:
Should this line be: #if defined(AUX3_SERIAL) ?

As AUX3_SERIAL_TX is not defined elsewhere